programs/conditional_vault/src/instructions/add_metadata_to_conditional_tokens.rs

```diff
@@ -8,53 +8,41 @@ pub mod proph3t_deployer {
 
 #[derive(AnchorSerialize, AnchorDeserialize)]
 pub struct AddMetadataToConditionalTokensArgs {
-    pub proposal_number: u64,
-    pub on_finalize_uri: String,
-    pub on_revert_uri: String,
+    pub name: String,
+    pub symbol: String,
+    pub uri: String,
 }
 
 #[derive(Accounts)]
 pub struct AddMetadataToConditionalTokens<'info> {
     #[account(mut)]
     pub payer: Signer<'info>,
-    #[account(
-        mut,
-        has_one = underlying_token_mint,
-    )]
-    pub vault: Account<'info, ConditionalVault>,
     #[account(mut)]
-    pub underlying_token_mint: Account<'info, Mint>,
-    pub underlying_token_metadata: Account<'info, MetadataAccount>,
-    #[account(
-        mut,
-        mint::authority = vault,
-        mint::freeze_authority = vault,
-        mint::decimals = underlying_token_mint.decimals
-    )]
-    pub conditional_on_finalize_token_mint: Account<'info, Mint>,
+    pub vault: Account<'info, ConditionalVault>,
```

Comment:
How come the vault no longer has to be connected to the conditional token mint via a `has_one` or other check?

Reply:
the `mint::authority = vault,` check on `conditional_token_mint` is probably enough maybe

Reply:
these checks are pretty pointless until we're able to verify URIs on-chain so that we can make this permissionless